### PR TITLE
feat(docker): Honor the host specified in current docker context

### PR DESCRIPTION
**Pull request by rajiv-k**
One can have multiple docker hosts that they manage and can switch between them using the "docker context" command.
Previously, we used to connect to either the "default docker host" (for example, `unix:///var/run/docker.sock` in case of Linux/macOS) or to the host specified via the `"DOCKER_HOST"` environment variable. However, if one switched to a different context, we had no way of detecting that.

Now, we determine the docker host that we connect to by evaluating the following sources in decreasing order of precedence:
- value of `"DOCKER_HOST"` environment variable
- host specified in the context set via the `"DOCKER_CONTEXT"` environment variable
- "default docker host" as applicable, otherwise

Fixes: #285

### CODE REVIEW

File: pkg/commands/docker.go [hunk 1]
```diff
@@ -314,3 +323,67 @@ func (c *DockerCommand) DockerComposeConfig() string {
...
+	}
+
+	if currentContext == "" {
+		// If a docker context is neither specified via the "DOCKER_CONTEXT" environment variable nor via the
+		// $HOME/.docker/config file, then we fall back to connecting to the "default docker host" meant for
+		// the host operating system.
+		return defaultDockerHost, nil
+	}
+
+	storeConfig := ctxstore.NewConfig(
```
[the remark this answers is not in the document]
Comment (rajiv-k) [reply pkg/commands/docker.go:359]: Sure! Here's the gist of what's happening.

Line: 364: We create a new `Store` by passing the path to store directory (`$HOME/.docker/contexts`) and the "store config" (which is used to return a typed value from a "generic" value that the `Store` can _store_). 
Here, we are interested in retrieving the information associated with a `docker context` aka the `EndpointMeta`, which contains the docker host information. That is why the "store config" includes a `getter` meant for the `EndpointMeta` (Line: 359).

The rest of the surrounding lines are just plumbing code to extract the host information and error handling.

### Bonus:
#### What's inside the `contexts` directory?
```sh
$ tree ~/.docker/contexts/meta/
$HOME/.docker/contexts/meta/
├── 2c26b46b68ffc68ff99b453c1d30413413422d706483bfa0f98a5e886266e7ae
│   └── meta.json
├── ed027fa4e9dc0950c3d3acf4a7bbaf242810ff73290de387e34e6d26c2a5b21b
│   └── meta.json
├── f24fd3749c1368328e2b149bec149cb6795619f244c5b584e844961215dadd16
│   └── meta.json
└── fe9c6bd7a66301f49ca9b6a70b217107cd1284598bfc254700c989b916da791e
    └── meta.json
```

#### What's inside the `meta.json`?

```sh
$ cat $HOME/.docker/contexts/meta/f24fd3749c1368328e2b149bec149cb6795619f244c5b584e844961215dadd16/meta.json | jq '.'
{
  "Name": "colima",
  "Metadata": {
    "Description": "colima"
  },
  "Endpoints": {
    "docker": {
      "Host": "unix:///$HOME/.colima/default/docker.sock",
      "SkipTLSVerify": false
    }
  }
}
```

**Comment by rajiv-k** [comment]
@lclpedro Creating the symlink would work. But the issue is that if you manage multiple docker environments, you will have to re-create these symlinks whenever you want to target a different environment. `docker context` exists to solve this very problem.

**Comment by jesseduffield** [comment]
Sorry this took me so long to get back to @rajiv-k . Great work!

**Comment by whobutsb** [comment]
Thank you for adding this feature!  I added the feature request in December 2021.  Really happy to see it get added.

**Comment by whobutsb** [comment]
@rajiv-k - I was testing out this new functionality.  Is honoring the docker context only available for `http` connections?  Am I able to use `ssh` connections?  When I changed my docker context and ran lazydocker I wasn't able to connect.  Here is the inspection on my docker context: 
```

**Comment by rajiv-k** [comment]
@whobutsb Looks like a possible regression. I am working on a PR to make this work (again).

**Pull request merged**